File: CSharpSDK/Beans/AirdInfo.cs
﻿/*
 * Copyright (c) 2020 CSi Studio
 * AirdSDK and AirdPro are licensed under Mulan PSL v2.
 * You can use this software according to the terms and conditions of the Mulan PSL v2. 
 * You may obtain a copy of Mulan PSL v2 at:
 *          http://license.coscl.org.cn/MulanPSL2 
 * THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT, MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.  
 * See the Mulan PSL v2 for more details.
 */

using System;
using System.Collections.Generic;

namespace AirdSDK.Beans
{
    public class AirdInfo
    {
        /**
        * Aird version
        * Aird的版本号
        */
        public string version = "2.0.0";

        /**
         * Aird Code
         * Aird的版本编码
         */
        public int versionCode = 5;

        /**
       * [Core Field]
       * The array compressor strategy
       * [核心字段]
       * 数组压缩策略
       */
        public List<Beans.Compressor> compressors;

        /**
         * Instrument information list
         */
        public List<Instrument> instruments;

        /**
         * dataProcessing information list
         */
        public List<DataProcessing> dataProcessings;

        /**
         * The processed software information
         * 处理的软件信息
         */
        public List<Software> softwares;

        /**
         * The file information before converting
         * 处理前的文件信息
         */
        public List<ParentFile> parentFiles;

        /**
         * [Core Field]
         * Store the window rangs which have been adjusted with experiment overlap
         * [核心字段]
         * 存储SWATH窗口信息,窗口已经根据overlap进行过调整
         */
        public List<WindowRange> rangeList;

        /**
         * [Core Field]
         * Used for block index(PRM/DIA/ScanningSWATH/DDA)
         * When using for storing SWATH window range information, the swath windows have already been adjusted by overlap between windows.
         * [核心字段]
         * 用于存储Block的索引（适用于PRM/DIA/ScanningSwath/DDA）
         * 当存储SWATH窗口信息,窗口已经根据overlap进行过调整
         */
        public List<BlockIndex> indexList;

        /**
         * [Core Field]
         * AirdType, Support for DIA/SWATH, PRM, DDA, SCANNING_SWATH, COMMON
         * [核心字段]
         * Aird支持的采集模式的类型,目前支持DIA,PRM,DDA, DDAPasef, DIAPasef, SCANNING_SWATH和COMMON 5种
         */
        public string type;

        /**
         * the vendor file size
         * 原始文件的文件大小,单位byte
         */
        public long fileSize;

        /**
         * the total spectra count
         * 总计拥有的光谱数
         */
        public long totalCount;

        /**
        * the aird file path.
        * 转换压缩后的aird二进制文件路径,默认读取同目录下的同名文件,如果不存在才去读本字段对应的路径
        */
        public string airdPath;

        /**
        * See ActivationMethod: HCD,CID....
        * 碎裂方法,详情见ActivationMethod类
        */
        public string activator;

        /**
        * Activation Method Energy
        * 轰击能量
        */
        public float energy;

        /**
        * Profile or Centroid
        */
        public string msType;

        /**
         * rt unit
         * rt的时间单位, Aird格式下统一为秒
         */
        public string rtUnit = "second";

        /**
         * polarity
         */
        public string polarity;

        /**
         * If ignore the point which intensity = 0
         * 是否忽略intensity为0的点
         */
        public bool ignoreZeroIntensityPoint = true;

        /**
         * Use in ion mobility acquisition method
         * 如果是Mobility采集模式,本字段会启用
         */
        public MobiInfo mobiInfo;

        /**
        * the aird file creator
        * 实验的创建者
        */
        public string creator;

        /**
        * the features. See Features.cs
        * 特征键值对,详情见Features.cs
        */
        public string features;

        /**
        * the create data
        * 文件的创建日期
        */
        public string createDate;
    }
}

File: PythonSDK/Beans/AirdInfo.py
from Beans.BlockIndex import BlockIndex
from Beans.Compressor import Compressor
from Beans.DataProcessing import DataProcessing
from Beans.Instrument import Instrument
from Beans.MobiInfo import MobiInfo
from Beans.ParentFile import ParentFile
from Beans.Software import Software
from Beans.WindowRange import WindowRange


class AirdInfo:
    def __init__(self, dict):
        self.version = dict['version'] if 'version' in dict else None
        self.versionCode = dict['versionCode'] if 'versionCode' in dict else None
        compressors = []
        if "compressors" in dict and len(dict['compressors']) > 0:
            for compressorDict in dict['compressors']:
                compressors.append(Compressor(compressorDict))
        self.compressors = compressors

        instruments = []
        if "instruments" in dict and len(dict['instruments']) > 0:
            for instrumentDict in dict['instruments']:
                instruments.append(Instrument(instrumentDict))
        self.instruments = instruments

        dataProcessings = []
        if "dataProcessings" in dict and len(dict['dataProcessings']) > 0:
            for dataProcessingDict in dict['dataProcessings']:
                dataProcessings.append(DataProcessing(dataProcessingDict))
        self.dataProcessings = dataProcessings

        softwares = []
        if "softwares" in dict and len(dict['softwares']) > 0:
            for softwareDict in dict['softwares']:
                softwares.append(Software(softwareDict))
        self.softwares = softwares

        parentFiles = []
        if "parentFiles" in dict and len(dict['parentFiles']) > 0:
            for parentFileDict in dict['parentFiles']:
                parentFiles.append(ParentFile(parentFileDict))
        self.parentFiles = parentFiles

        rangeList = []
        if "rangeList" in dict and len(dict['rangeList']) > 0:
            for rangeDict in dict['rangeList']:
                rangeList.append(WindowRange(rangeDict))
        self.rangeList = rangeList

        indexList = []
        if "indexList" in dict and len(dict['indexList']) > 0:
            for indexDict in dict['indexList']:
                indexList.append(BlockIndex(indexDict))
        self.indexList = indexList

        self.type = dict['type'] if 'type' in dict else None
        self.fileSize = dict['fileSize'] if 'fileSize' in dict else None
        self.totalCount = dict['totalCount'] if 'totalCount' in dict else None
        self.airdPath = dict['airdPath'] if 'airdPath' in dict else None
        self.activator = dict['activator'] if 'activator' in dict else None
        self.energy = dict['energy'] if 'energy' in dict else None
        self.msType = dict['msType'] if 'msType' in dict else None
        self.rtUnit = dict['rtUnit'] if 'rtUnit' in dict else None
        self.polarity = dict['polarity'] if 'polarity' in dict else None
        self.ignoreZeroIntensityPoint = dict['ignoreZeroIntensityPoint'] if 'ignoreZeroIntensityPoint' in dict else None
        if 'mobiInfo' in dict:
            self.mobiInfo = MobiInfo(dict['mobiInfo'])
        self.creator = dict['creator'] if 'creator' in dict else None
        self.features = dict['features'] if 'features' in dict else None



